### PR TITLE
add HEVC(H.265) codec to iOS,Mac VideoWriter

### PR DESCRIPTION
**Pull request by Varvrar**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/videoio/src/cap_avfoundation.mm [hunk 1]
```diff
@@ -1255,13 +1255,15 @@ -(int) updateImage {
...
-    // Two codec supported AVVideoCodecH264 AVVideoCodecJPEG
+    // Three codec supported AVVideoCodecH264 AVVideoCodecJPEG AVVideoCodecTypeHEVC
     // On iPhone 3G H264 is not supported.
     if (fourcc == CV_FOURCC('J','P','E','G') || fourcc == CV_FOURCC('j','p','e','g') ||
             fourcc == CV_FOURCC('M','J','P','G') || fourcc == CV_FOURCC('m','j','p','g') ){
         codec = [AVVideoCodecJPEG copy]; // Use JPEG codec if specified, otherwise H264
     }else if(fourcc == CV_FOURCC('H','2','6','4') || fourcc == CV_FOURCC('a','v','c','1')){
             codec = [AVVideoCodecH264 copy];
+    }else if(fourcc == CV_FOURCC('H','E','V','C') || fourcc == CV_FOURCC('h','e','v','c') || fourcc == CV_FOURCC('H','2','6','5')){
+            codec = [AVVideoCodecTypeHEVC copy];
```
Comment (alalek) [inline modules/videoio/src/cap_avfoundation.mm:1266]: Please add proper version check for that.

**Review by alalek — COMMENTED** [review]
Thank you for contribution!

File: modules/videoio/src/cap_avfoundation.mm (hunk 1 above)
Comment (Varvrar) [reply modules/videoio/src/cap_avfoundation.mm:1266]: Sorry, version of what should I check? iOS?

**Comment by VadimLevin** [comment]
I added required versions checks based on the availability of the HEVC key in the `AVVideoSettings.h:43`:
```
AVF_EXPORT AVVideoCodecType const AVVideoCodecTypeHEVC /* @"hvc1" */                    API_AVAILABLE(macos(10.13), ios(11.0), tvos(11.0)) API_UNAVAILABLE(watchos);
```

@alalek Can you proceed with review, please?

File: modules/videoio/src/cap_avfoundation.mm (hunk 1 above)
Comment (Varvrar) [reply modules/videoio/src/cap_avfoundation.mm:1271]: Thanks to @VadimLevin

**Review by alalek — APPROVED** [review]

**Pull request merged**